src/main/java/com/elisis/gtnhlanth/common/tileentity/DissolutionTank.java

```diff
@@ -87,96 +87,31 @@ private boolean addGlass(Block block, int meta) {
 
     @Override
     public boolean checkRecipe(ItemStack itemStack) {
-
         // GT_Log.out.print("in checkRecipe");
 
         ArrayList<FluidStack> tFluidInputs = this.getStoredFluids();
         FluidStack[] tFluidInputArray = tFluidInputs.toArray(new FluidStack[0]);
         ItemStack[] tItems = this.getStoredInputs().toArray(new ItemStack[0]);
         long tVoltage = this.getMaxInputVoltage();
 
-        // Collection<GT_Recipe> tRecipes = RecipeAdder.instance.DigesterRecipes.mRecipeList;
         GT_Recipe tRecipe = RecipeAdder.instance.DissolutionTankRecipes.findRecipe(
                 getBaseMetaTileEntity(), false, tVoltage, tFluidInputArray, tItems);
 
-        if (tRecipe == null) return false;
-
+        if (tRecipe == null || !tRecipe.isRecipeInputEqual(true, tFluidInputArray, tItems)) return false;
         // GT_Log.out.print("Recipe not null\n");
-        if (tRecipe.isRecipeInputEqual(true, tFluidInputArray, tItems)) {
-
-            this.mEfficiency = (10000 - (this.getIdealStatus() - this.getRepairStatus()) * 1000);
-            this.mEfficiencyIncrease = 10000;
-            this.calculateOverclockedNessMulti(tRecipe.mEUt, tRecipe.mDuration, 1, tVoltage);
-
-            if (mMaxProgresstime == Integer.MAX_VALUE - 1 && this.mEUt == Integer.MAX_VALUE - 1) return false;
-
-            if (this.mEUt > 0) this.mEUt = (-this.mEUt);
-
-            FluidStack majorGenericFluid = tRecipe.mFluidInputs[0];
-            FluidStack minorGenericFluid = tRecipe.mFluidInputs[1];
-
-            FluidStack majorInput = null;
-            FluidStack minorInput = null;
-
-            int majorAmount = 0;
-            int minorAmount = 0;
-
-            FluidStack fluidInputOne = tFluidInputs.get(0);
-            FluidStack fluidInputTwo = tFluidInputs.get(1);
 
-            majorInput = ((fluidInputOne.getUnlocalizedName().equals(majorGenericFluid.getUnlocalizedName()))
-                    ? fluidInputOne
-                    : fluidInputTwo);
-            // GT_Log.out.print(majorInput.getLocalizedName());
-            if (fluidInputOne.getUnlocalizedName().equals(majorGenericFluid.getUnlocalizedName())) {
-                if (fluidInputTwo.getUnlocalizedName().equals(minorGenericFluid.getUnlocalizedName())) {
-                    majorInput = fluidInputOne;
-                    majorAmount = fluidInputOne.amount;
-                    minorInput = fluidInputTwo;
-                    minorAmount = fluidInputTwo.amount;
```

Comment:
You're removing all of the ratio functionality of the Dissolution Tank here, as far as I can tell. This code isn't all redundant.

Reply:
Oh sorry, I didn't read logic carefully and didn't notice `mSpecialValue` actually being used.